COMET/misc_plugins/PlotScripts/analysis_scripts/IV_PQC.py

```diff
@@ -1,82 +1,670 @@
+        try:
+            ynew = f(xnew)
+        except Exception:
+            ynew = np.arange(len(list(xnew)))
+
+        return xnew, ynew
+
+    def add_single_plots(self, xaxis, yaxis, name):
+        points_plot = (xaxis, yaxis)
+        interp_plot = hv.Curve(points_plot)
+        interp_plot = customize_plot(interp_plot, name, self.config["IV_PQC"])
+
+        return interp_plot
+
+    def interpol(self, df, xaxis, yaxis):
+        xnew, ynew = self.interpolated_axis(df, xaxis, yaxis)
+        capa_interp_plot = self.add_single_plots(xnew, ynew, "Capacity")
+
+        # Build derivatives of the interpolated data
+        firstdev_interp = self.build_first_derivative(xnew, ynew)
+        derivative_interpolation_plot = self.add_single_plots(xnew, firstdev_interp, "derivative")
+        seconddev_interp = self.build_second_derivative(xnew, ynew)
+        secondderivative_interp_plot = self.add_single_plots(xnew, seconddev_interp, "derivative2")
+
+        # Find the flatband-voltage through the maximum value of the first derivative
+        item_max = firstdev_interp.argmax()
+        voltage_value_of_max_firstder = xnew[item_max]
+        max_firstder_plot = hv.VLine(voltage_value_of_max_firstder).opts(line_width=1.0)
+        fbvoltage_firstderivative.append(voltage_value_of_max_firstder)
+
+        return capa_interp_plot, derivative_interpolation_plot, secondderivative_interp_plot, max_firstder_plot, voltage_value_of_max_firstder
+
+    def interp_derivative_diode(self, df, xaxis, yaxis):
+        # Interpolate the capacity
+        xnew, ynew = self.interpolated_axis(df, xaxis, yaxis)
+        Onec2 = 1 / (ynew * ynew)  # 1C2 array
+        capacity_curve = self.add_single_plots(xnew, Onec2, "1C2")
+
+        # Derivative
+        first_dev = self.build_first_derivative(xnew, Onec2)
+        derivative_onec2_curve = self.add_single_plots(xnew, first_dev, "1C2")
+
+        # Savgolay plot
+        derivative_savgol_filtered = scipy.signal.savgol_filter(first_dev, 5, 3)  # Window size 5, polynomial order 3
+        deronec2_savgol_plot = self.add_single_plots(xnew, derivative_savgol_filtered, "1C2")
+
+        return capacity_curve, derivative_onec2_curve, deronec2_savgol_plot
+
+    def find_flatBand_voltage(self, plot, data, configs, indexMax, indexMin, df, cv_files, voltage_value_of_max_firstder, **addConfigs): #cv is the list containing all the cvmos files
+        """
+        Finds the full depletion voltage of all data samples and adds a vertical line for the full depletion in the
+        plot. Vertical line is the mean of all measurements. Furthermore, adds a text with the statistics.
+        :param plot: The plot object
+        :param data: The data files
+        :param configs: the configs
+        :param **addConfigs: the configs special for the 1/C2 plot, it is recommended to pass the same options here again, like in the original plot!
+        :return: The updated plot
+        """
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Global variable 'fit_stats' undefined at the module level](https://app.codacy.com/manual/dominic.bloech/UniDAQ/pullRequest?prid=5481807)